contracts/network/network.cpp

```diff
@@ -0,0 +1,124 @@
+#include "network.hpp"
+
+void network::open(vaccount addr, eosio::extended_symbol symbol, eosio::name payer) {
+  account_table acc_tbl(_self, _self.value);
+  uint64_t id = acc_tbl.available_primary_key();
+  eosio::extended_asset asset(0, symbol);
+
+  // check user does not already have a balance for this symbol
+  auto acc_tbl_idx = acc_tbl.get_index<"token"_n>();
+  auto idx_key = make_token_index(symbol.get_contract(), addr);
+  auto itr_start = acc_tbl_idx.lower_bound(idx_key);
+  auto itr_end = acc_tbl_idx.upper_bound(idx_key);
+
+  bool found = false;
+  for (; itr_start != itr_end; ++itr_start) {
+    found = true;
+    break;
+  }
+
+  if (found == false) {
+    acc_tbl.emplace(payer,
+                    [&](auto& a)
+                    {
+                      a.id = id;
+                      a.nonce = 0;
+                      a.address = addr;
+                      a.balance = asset;
+                    });
+  }
+}
+
+void network::transfer_handler(name from, name to, asset quantity, std::string memo) {
```

Comment:
Add transfer_handler functionality (deposit into your account when memo is your account id) to README